common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp

```diff
@@ -738,18 +738,20 @@ bool mon_wlan_hal_dwpal::update_stations_stats(const std::string &vap_iface_name
         }
     }
 
-    //save avarage RSSI in watt
-    for (uint8_t i = 0; i < numOfValidArgs[5]; i++) {
-        float s_float = float(beerocks::string_utils::stoi(std::string(ShortTermRSSIAverage[i])));
+    //save average RSSI in watt
+    // ShortTermRSSIAverage values are space-separated:
+    for (const auto &rssi: beerocks::string_utils::str_split(ShortTermRSSIAverage, ' ')) {
+        float s_float = float(beerocks::string_utils::stoi(std::string(rssi)));
         if (s_float > beerocks::RSSI_MIN) {
             sta_stats.rx_rssi_watt += std::pow(10, s_float / float(10));
             sta_stats.rx_rssi_watt_samples_cnt++;
         }
     }
 
-    //save avarage SNR in watt
-    for (uint8_t i = 0; i < numOfValidArgs[6]; i++) {
-        float s_float = float(beerocks::string_utils::stoi(std::string(SNR[i])));
+    //save average SNR in watt
```

Comment:
This comment from the old code is also incorrect, it's a ratio (just a number), it was probably meant to say it's converted from logarithmic to dynamic scale.